### PR TITLE
change behaviour when applying firewall rules

### PR DESCRIPTION
**Pull request by acalm**
instead of running firewall apply for a specific firewall asset, this change runs the firewall_apply function for all assets where asset_subtype is set to firewall and url is set

### CODE REVIEW

File: socrates_api/serializers.py [hunk 1]
```diff
@@ -652,30 +652,9 @@ class Meta(RethinkSerializer.Meta):
...
-                        'network': {'device': domain},
-                    }))
-                    break
-                except:
-                    pass
-            else:
-                continue
-            firewall_apply.apply_async((asset,))
-        return ruleset
+    def update(self):
```
Comment (dhozac) [inline socrates_api/serializers.py]: Arguments still need to be present here.

File: socrates_api/tasks.py [hunk 2]
```diff
@@ -1366,6 +1366,13 @@ def firewall_apply(asset):
         return firewall_apply_ansible(asset, url, networks, address_groups, rulesets)
     return False
 
+@shared_task
+def firewall_apply_all():
+    for asset in AssetSerializer.filter({'asset_subtype': 'firewall'}):
+        if asset.get('url'):
+            firewall_apply.apply_async((asset,))
+    return False
```
Comment (dhozac) [inline socrates_api/tasks.py]: This can probably be removed.

File: socrates_api/serializers.py [hunk 3]
```diff
@@ -652,30 +652,9 @@ class Meta(RethinkSerializer.Meta):
             ('permissions_write', r.row['permissions']['write'], {'multi': True}),
         ]
 
-    def update(self, instance, data):
-        from socrates_api.tasks import firewall_apply
-        ruleset = super(FirewallRuleSetSerializer, self).update(instance, data)
```
Comment (dhozac) [inline socrates_api/serializers.py:657]: This is required.

**Review by dhozac — CHANGES_REQUESTED** [review]

File: socrates_api/serializers.py (hunk 3 above)
Comment (acalm) [reply socrates_api/serializers.py:657]: was a bit to eager with the scissors, should be fixed now

**Pull request merged**